mula/scheduler/models/schedule.py

```diff
@@ -30,14 +30,41 @@ class Schedule(BaseModel):
 
     tasks: list[Task] = []
 
-    deadline_at: datetime | None = None
+    _deadline_at: datetime | None = None
     created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
     modified_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
 
+    def __init__(self, **data):
+        super().__init__(**data)
+
+        if "deadline_at" in data:
+            self.deadline_at = data["deadline_at"]
+        elif "deadline_at" not in data and self.schedule is not None:
+            self.deadline_at = cron.next_run(self.schedule)
```

Comment:
This seems more like a [model_post_init](https://docs.pydantic.dev/latest/api/base_model/#pydantic.BaseModel.model_post_init) functionality, since there's no validation here anymore and the object is already initialized. Consider using that, since you rarely need to use `__init__` here nowadays

Reply:
Using your suggestion, was able to create a simpler implementation 🙏